File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        // if(head==NULL) return head;
        // ListNode *p=head;
        // ListNode *s,*r;
        // s=NULL;
        // while(p->next){
        //     r=p->next;
        //     p->next=s;
        //     s=p;
        //     p=r;
        // }
        // p->next=s;
        // return p;
        ListNode *s=NULL;
        ListNode *p=head;
        while(p){
            ListNode *temp=p->next;
            p->next=s;
            s=p;
            p=temp;
        }
        return s;
    }
};

File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode s=null;
        ListNode p=head;
        while(p!=null){
            ListNode temp=p.next;
            p.next=s;
            s=p;
            p=temp;
        }
        return s;
    }
}